core/server/master/src/main/java/alluxio/master/meta/MetaMasterSync.java

```diff
@@ -143,6 +131,29 @@ private void handleCommand(MetaCommand cmd) throws IOException {
     }
   }
 
+  /**
+   * Sets the master id and registers with the Alluxio leader master.
```

Comment:
NIT: do we register with master first before set the master id? if yes, I would call this method `registerAndSetId` and update the javadoc